compiler/rustc_mir/src/interpret/intrinsics/caller_location.rs

```diff
@@ -91,7 +91,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             .type_of(self.tcx.require_lang_item(LangItem::PanicLocation, None))
             .subst(*self.tcx, self.tcx.mk_substs([self.tcx.lifetimes.re_erased.into()].iter()));
         let loc_layout = self.layout_of(loc_ty).unwrap();
-        let location = self.allocate(loc_layout, MemoryKind::CallerLocation);
+        let location = self.allocate(loc_layout, MemoryKind::CallerLocation)?;
```

Comment:
I think we should unwrap here instead of at the callers to this

Reply:
this is the only non-panicking failure here, and it is always caught to panic

Reply:
`InterpCx::emulate_intrinsic` calls `alloc_caller_location_for_span` which in turn calls this function, and `InterpCx::emulate_intrinsic` doesn't unwrap the `InterpResult`

Reply:
it should though, right? same reasoning as at the actually unwrapping site